integreat_compass/cms/views/index.py

```diff
@@ -1,9 +1,124 @@
+            },
+        )
+    
+    def post(self, request):
+        """
+        This functions filters
+
+        :param request: The current request
+        :type request: ~django.http.HttpRequest
+
+        :return: The rendered template response
+        :rtype: ~django.template.response.TemplateResponse
+        """
+        offers = Offer.objects.all()
+        tags = Tag.objects.all()
+
+        selected_filters = {
+            "tags" : request.POST.getlist("offer-tags"),
+            "mode_type" : request.POST.get("mode-type"),
+            "group_type" : request.POST.get("group-type"),
+            "search" : request.POST.get("offer-search"),
+        }
+        
+        filtered_offers = []
+        selected_filters["tags"] = [eval(i) for i in selected_filters["tags"]]
+
+
+        for offer in offers:
+            tag_ids = [tag.id for tag in offer.tags.all()]
+
+            if any(item in selected_filters["tags"] for item in tag_ids):
+                filtered_offers.append(self.store_offer_data_in_dict(offer))
+                continue
+            
+            """
+            if selected_filters["mode_type"] == offer.mode_type:
+                filtered_offers.append(self.store_offer_data_in_dict(offer))
+                continue
+            
+            
+            if selected_filters["group_type"] == offer.group_type:
+                filtered_offers.append(self.store_offer_data_in_dict(offer))
+                continue
+            
+            if re.search(selected_filters["search"], offer.versions.first().title, re.IGNORECASE):
+                filtered_offers.append(self.store_offer_data_in_dict(offer))
+                continue
+            """
+
+        amount_of_offers = len(filtered_offers)
+        return render(
+            request,
+            self.template_name,
+            {
+                "offers" : filtered_offers,
+                "tags" : tags,
+                "amount_of_offers" : amount_of_offers,
+                "offer_group_types" : offer_group_types.CHOICES,
+                "offer_mode_types" : offer_mode_types.CHOICES
+            },
+        )
+    
+    def store_offer_data_in_dict(self, offer):
+        """
+        This function builds the dict that is later rendered by the template
+
+        :param offer: The current request
+        :type offer: ~django.http.HttpRequest
+
+        :return: Dict that contains the relevant data that's supposed to be rendered by the template
+        :rtype: dict
+
+        """
+        offer_dict = {}
+        offer_dict["image"] = offer.versions.first().title_image.url
+        offer_dict["title"] = offer.versions.first().title
+        offer_dict["organization"] = offer.organization.name
+        offer_dict["description"] = offer.versions.first().description
+        offer_dict["creation_date"] = offer.versions.first().offer_version_date
+        offer_dict["tags"] = offer.tags.all()
+        return offer_dict
+    
```

Comment:
With the current implementation we would need also a "reset filter" button. I don't mind that, but if there is a better solution it would be nice if we could circumvent it